diff --git a/backends/cxxrtl/cxxrtl_backend.cc b/backends/cxxrtl/cxxrtl_backend.cc
index 8dc14863d..532d57818 100644
--- a/backends/cxxrtl/cxxrtl_backend.cc
+++ b/backends/cxxrtl/cxxrtl_backend.cc
@@ -1417,8 +1417,8 @@ struct CxxrtlWorker {
 				RTLIL::SigBit clk_bit = cell->getPort(ID::CLK)[0];
 				clk_bit = sigmaps[clk_bit.wire->module](clk_bit);
 				if (clk_bit.wire) {
-					f << indent << "if (" << (cell->getParam(ID::CLK_POLARITY).as_bool() ? "posedge_" : "negedge_")
-					            << mangle(clk_bit) << ") {\n";
+					f << indent << "if (this->" << (cell->getParam(ID::CLK_POLARITY).as_bool() ? "posedge_" : "negedge_")
+					            << mangle(clk_bit) << "()) {\n";
 				} else {
 					f << indent << "if (false) {\n";
 				}
@@ -1802,11 +1802,13 @@ struct CxxrtlWorker {
 			if (i != 0)
 				f << " || ";
 
+			f << "this->";
 			if (trg_polarity[i] == State::S1)
 				f << "posedge_";
 			else
 				f << "negedge_";
 			f << mangle(trg_bit);
+			f << "()";
 		}
 		f << ") {\n";
 		inc_indent();
@@ -1838,8 +1840,8 @@ struct CxxrtlWorker {
 			RTLIL::SigBit clk_bit = port.clk[0];
 			clk_bit = sigmaps[clk_bit.wire->module](clk_bit);
 			if (clk_bit.wire) {
-				f << indent << "if (" << (port.clk_polarity ? "posedge_" : "negedge_")
-					    << mangle(clk_bit) << ") {\n";
+				f << indent << "if (this->" << (port.clk_polarity ? "posedge_" : "negedge_")
+					    << mangle(clk_bit) << "()) {\n";
 			} else {
 				f << indent << "if (false) {\n";
 			}
@@ -1984,8 +1986,8 @@ struct CxxrtlWorker {
 				RTLIL::SigBit clk_bit = port.clk[0];
 				clk_bit = sigmaps[clk_bit.wire->module](clk_bit);
 				if (clk_bit.wire) {
-					f << indent << "if (" << (port.clk_polarity ? "posedge_" : "negedge_")
-					            << mangle(clk_bit) << ") {\n";
+					f << indent << "if (this->" << (port.clk_polarity ? "posedge_" : "negedge_")
+					            << mangle(clk_bit) << "()) {\n";
 				} else {
 					f << indent << "if (false) {\n";
 				}
@@ -2169,6 +2171,7 @@ struct CxxrtlWorker {
 		inc_indent();
 			f << indent << "bool converged = " << (eval_converges.at(module) ? "true" : "false") << ";\n";
 			if (!module->get_bool_attribute(ID(cxxrtl_blackbox))) {
+#if 0
 				for (auto wire : module->wires()) {
 					if (edge_wires[wire]) {
 						for (auto edge_type : edge_types) {
@@ -2185,6 +2188,7 @@ struct CxxrtlWorker {
 						}
 					}
 				}
+#endif
 				for (auto wire : module->wires())
 					dump_wire(wire, /*is_local=*/true);
 				for (auto node : schedule[module]) {
diff --git a/passes/cmds/exec.cc b/passes/cmds/exec.cc
index e5fa4fb41..e9cc34b30 100644
--- a/passes/cmds/exec.cc
+++ b/passes/cmds/exec.cc
@@ -141,7 +141,7 @@ struct ExecPass : public Pass {
 					}
 
 				} else
-					log_cmd_error("Unknown option \"%s\" or \"--\" doesn\'t precede command.", args[argidx].c_str());
+					log_cmd_error("Unknown option \"%s\" or \"--\" doesn\'t precede command.\n", args[argidx].c_str());
 			}
 		}
 
